src/lib.rs

```diff
@@ -245,13 +248,15 @@ pub mod pallet {
 			let sindex = Self::get_supersig_index_from_id(&supersig_id)
 				.ok_or(Error::<T>::SupersigNotFound)?;
 
-			if !Self::is_user_in_supersig(sindex, &who) {
-				return Err(Error::<T>::NotMember.into())
-			}
 			let nonce = Self::nonce_call(sindex);
 			let data = call.encode();
-			let deposit = <BalanceOf<T>>::from(data.len() as u32)
-				.saturating_mul(T::PreimageByteDeposit::get());
+
+			if Calls::<T>::iter_prefix_values(sindex).any(|elem| elem.data == data) {
+				return Err(Error::<T>::CallAlreadyExists.into())
+			}
```

Comment:
We are never supposed to loop in substrate. This can be really bad for performances.
Do you have an idea on how we can do instead ?